esi_leap/api/controllers/v1/node.py

```diff
@@ -64,15 +64,26 @@ def __init__(self, **kwargs):
 
 class NodesController(rest.RestController):
 
-    @wsme_pecan.wsexpose(NodeCollection)
-    def get_all(self):
+    @wsme_pecan.wsexpose(NodeCollection, wtypes.text,
+                         wtypes.text, wtypes.text)
+    def get_all(self, resource_class=None, owner=None,
+                lessee=None):
         context = pecan.request.context
 
+        filter_args = {
+            'resource_class': resource_class,
+            'owner': owner,
+            'lessee': lessee
```

Comment:
This won't quite work, because ESI allows users to enter either the project name or the project UUID; however, the Ironic API expects only the UUID. You'll need to convert similarly to how we do it for leases: https://github.com/CCI-MOC/esi-leap/blob/master/esi_leap/api/controllers/v1/lease.py#L103